### ISSUE TITLE
The account will be registered after removing the app from recent

### ISSUE THREAD

**Issue by RajanVindaloo** [BelledonneCommunications/linphone-android#912]
Hello there. 
I am using Linphone 4.1 version. 
I recently found that my account is still registered after removing the app from the recent app list in Linphone 4.2 version. That's a good thing with the application. I am happy with that and furthermore some personal reasons I cannot update my app to Linphone 4.2. 
I did not find where is the exact difference in the code. Would you please help me with that?

**Comment by Viish** [comment]
Hi,
I'm not sure I understand your question.
You want to know how when the app is killed the account doesn't unregister?
If yes, simply call setNetworkReachable(false) on the Core.
If no, please explain a bit more.

**Comment by RajanVindaloo** [comment]
"You want to know how when the app is killed the account doesn't unregister?"
The exact thing I want to know is this. I've put setNetworkReachable(false) in LinphoneService and LinphoneManager as it is set in Linphone Version 4.2. But, I am not able to receive call or message.

**Comment by Viish** [comment]
Yes this is expected as setNetworkReachable will prevent the app from unregistering but also from accessing the network unless you call setNetworkReachable(true).
If you haven't done so I strongly recommend you set the [sip] auto_net_state_mon=1 in your config file so the SDK will manage automatically the network.

**Comment by RajanVindaloo** [comment]
[sip] auto_net_state_mon=1 is already set in linphonerc_factory. Still not working.
If I've to put setNetworkReachable(true) into some file then please let me know.

**Comment by Viish** [comment]
Normally you shouldn't have to call setNetworkReachable(true) if the only place you call setNetworkReachable(false) is right before the Core will be either stopped or destroyed.

**Comment by RajanVindaloo** [comment]
Okay. Then I've set all things as per your instruction. But still, I did not found my application registred. Please guide me further.

**Comment by Viish** [comment]
Only call setNetworkReachable(false) when your app is being switched away, that's all.
You can always call setNetworkReachable(true) when it's brought to foreground if you want to be sure but that shouldn't be necessary.

**Comment by RajanVindaloo** [comment]
The application is working fine in the foreground. But, my concern is with when the app is killed application is no more registered. I want that to work. I've followed your steps but still, the issue is there. I appreciate your help. I am in the same position where I was there before starting the conversation.

**Comment by Viish** [comment]
Check your logs to see if there is a REGISTER with EXPIRES=0 in the header.
If there is, you haven't call setNetworkReachable soon enough. If there isn't, then your account isn't registered as you want and you have another issue somewhere.

I won't help you anymore on this matter. You can always take a look at the source code of our app to see how we do it. 

Also we just integrated this feature in the SDK (master branch) for proxy configs with allow_push_notifications flag set to true (they won't unregister automatically when Core stops).

**Comment by RajanVindaloo** [comment]
Okay. Thanks for your help and time.
